lib/elixir/src/elixir_exp_clauses.erl

```diff
@@ -18,6 +18,10 @@ def(Fun, Args, Guards, Body, E) ->
 
 clause(Meta, Kind, Fun, {'->', ClauseMeta, [_, _]} = Clause, E) when is_function(Fun, 3) ->
   clause(Meta, Kind, fun(X, Acc) -> Fun(ClauseMeta, X, Acc) end, Clause, E);
+clause(Meta, 'cond', _Fun, {'->', _, [[{'_', _, Atom}], _]}, E) when is_atom(Atom) ->
```

Comment:
@whatyouhide use lists:last or in elixir_exp_clauses or in elixir_exp:expand({'cond', _, _}) (preferably the latter).